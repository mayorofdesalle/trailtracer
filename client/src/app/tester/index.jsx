import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
