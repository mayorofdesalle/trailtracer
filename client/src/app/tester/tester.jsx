import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };