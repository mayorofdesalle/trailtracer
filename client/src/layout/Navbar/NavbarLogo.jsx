import styled from 'styled-components';

import getImageUrl from '@utils/getImageUrl';
import Button from '@components/ui/Button';

/**
 * NavbarLogo
 * @description
 * This is a styled button that is used as the logo for the navbar.
 **/
const NavbarLogo = styled(Button)`
    border-radius: 0;
    width: clamp(6rem, min(12svw, 12svh), 8rem);
    background-color: ${({ theme }) => theme.colors.primary};
    -webkit-mask-image: url(${getImageUrl('logo')});
    mask-image: url(${getImageUrl('logo')});
    mask-clip: border-box;
    mask-size: contain;
    mask-repeat: no-repeat;
    mask-position: center;
    transition: 200ms ease-in-out;

    &:hover {
        border-radius: 0;
    }

    &:active {
        background-color: ${({ theme }) => theme.colors.secondary};
    }
`;

export default NavbarLogo;