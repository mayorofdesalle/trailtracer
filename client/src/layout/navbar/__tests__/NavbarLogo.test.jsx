import { describe, expect, it, render, screen } from '@app/tester';

import NavbarLogo from '../NavbarLogo';

describe('NavbarLogo', () => {

    it('should render correctly', () => {
        render(<NavbarLogo />);
        const logo = screen.getByRole('button');
        expect(logo).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<NavbarLogo />);
        expect(asFragment()).toMatchSnapshot();
    });

});