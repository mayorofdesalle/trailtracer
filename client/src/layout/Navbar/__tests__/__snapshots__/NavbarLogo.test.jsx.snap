// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavbarLogo > should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-beySPh sc-kFCroH zbkUW dBMJkV"
    type="button"
  />
</DocumentFragment>
`;
